### DISCUSSION TITLE
In the web client, the password saved by the account is suddenly not displayed

### DISCUSSION THREAD

**Discussion by refined-fish** [dani-garcia/vaultwarden#2428] [Q&A] [answered]
In the web client, the password saved by the account is suddenly not displayed, as shown below.
![image](https://user-images.githubusercontent.com/93998080/164443780-138d4b88-a1bb-48f8-add0-f2df9404dd02.png)
But in fact, this account has 278 passwords saved. As shown in the control panel.
![A@UVRX5 6POL}}F90C42IC6](https://user-images.githubusercontent.com/93998080/164444224-dc43bfa7-9db2-4448-a02b-5827550d5322.png)
At the same time, other clients keep prompting "sync failed", but the login operation can still be completed.
It all happened suddenly and I don't know what happened or how to troubleshoot. Hope someone can help me, thanks!

**Comment by Masgalor** [comment, marked as answer]
I think it is necessarry to give use more informations about your setup.
Assuming you're hosting vaultwarden yourself could you please fill the following list.

```
vaultwarden version:
Install method:
Clients used:
Reverse proxy and version:
MySQL/MariaDB or PostgreSQL version:
```

In additon to that some logs would be helpfull aswell.

**Reply by refined-fish** [reply]
thank for your help！
And sorry, I haven't noticed that this information has not been added.

vaultwarden version: 1.24.0
Install method: docker（Base: Debian-Synology3615xs）
Clients used: web/APP/browser plugin client
Reverse proxy and version: Nginx (Synology3615xs self-contained)
MySQL/MariaDB or PostgreSQL version: SQLite (the docker image self-contained)

**Reply by Masgalor** [reply]
Thank your adding those informations.

I may be wrong but it looks to me like there are some entries in your database but their content is somehow corrupted.

Did you work with organizations, or are all passwords owned by one user?

I am not an expert on SQLite but, the database itself might be damaged.
Do you have a backup of your database-file, or could you do an integrity-check of your existig one?

Finally, it would help to see the log entries produced after restarting the service.
Or put all existing logs in a gist and post the link here, so I can look there for errors.

**Reply by refined-fish** [reply]
I didn't use the organization function.
It's a pity I didn't backup the database before the error. But my other clients also have local cache to export. So actually my passwords are still there.

Sorry I don't quite understand what type of log you need. Below are the docker logs before and after the problem occurs.
[vaultwarden.csv](https://github.com/dani-garcia/vaultwarden/files/8531202/vaultwarden.csv)

Following are the supported strings for the control panel
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.24.0
* Web-vault version: v2.25.1
* Running within Docker: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.35.4
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** ADMIN_TOKEN


```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://****.*****.********.****:*****",
  "domain_origin": "*****://****.*****.********.****:*****",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 8,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_org_name": "XXXXXXXXX",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": true,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": true,
  "signups_verify_resend_limit": 3,
  "signups_verify_resend_time": 0,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": true,
  "smtp_from": "**************@***.***",
  "smtp_from_name": "XXXXXXXX",
  "smtp_host": "****.***.***",
  "smtp_password": "***",
  "smtp_port": XXX,
  "smtp_ssl": true,
  "smtp_timeout": 15,
  "smtp_username": "**************@***.***",
  "templates_folder": "data/templates",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

**Reply by Masgalor** [reply]
I see, the instructions were unclear.
But you got exactly what i wanted. ;)

```
[ERROR] thread 'unnamed' panicked at 'Error loading ciphers: DatabaseError(__Unknown, \database disk image is malformed\"")': src/db/models/cipher.rs:470"
```

Indicates a corrupted database I guess.

Again, I dont know a lot about sqlite, but you should backup your database now and lookup how to du integritychecks on sqlite databasefiles.

**Reply by refined-fish** [reply]
I agree with you and appreciate your help. 

But more importantly, I want to know how this problem came about and how to avoid it reappearing. I don't think I'll have such good luck every time. Maybe the next time the problem happens my other client doesn't save the complete data either. (&_&)

**Reply by Masgalor** [reply]
I do not know how to investigate any further in this case.

My advice is to use a database that is managed by an independet service e.g. mariadb.
After all sqlite is just a single file with direct writeacces for the client.
(Just my opinion, but you can educate yourself about the (dis-)advantages of sqlite.)

**Reply by refined-fish** [reply]
Because it is only for personal use, in order to avoid wasting resources, I have not used a separate database. But now I might have to reconsider. Thanks again for your kind help！

**Reply by Masgalor** [reply]
You are welcome.

May I ask you to mark this topic as answered, so others dont have to read through this just to notice that no more help is needed.

**Reply by BlackDex** [reply]
A sqlite file can break if you stop the application forcefully during a transaction for example.
Or the disk was full. Bad disk etc.. etc...

**Comment by BlackDex** [comment]
Did you checked the trash? They could be in there

**Reply by refined-fish** [reply]
Yes, I checked and the trash is completely empty too.

**Reply by refined-fish** [reply]
Actually it's not like the project was simply deleted. Because I created some folders and they are gone now. Like the first friend said, looks a lot like a read error.

**Reply by BlackDex** [reply]
Try to set the `LOG_LEVEL` to `debug` and also check the logs of the reverse proxy.
